### PR TITLE
Tilde compression for default prompt

### LINKED ISSUE #185: Tild compression for default prompt.
We shouldn't use colors in the default prompt (makes assumptions about the system we are running), but we could do tild compression if HOME is set.

### PR DESCRIPTION
**Pull request by x4lldux**
Resolves #185

### CODE REVIEW

**Comment by pauldub** [comment]
I think tilde compression should work for all paths prefixed with `$HOME`.o

Examples:

- `/home/foo` -> `~`
- `/home/foo/bar` -> `~/bar`

**Comment by x4lldux** [comment]
@pauldub good point! Fixed

**Comment by x4lldux** [comment]
@andrewchambers might this be some weird heisen-bug in job control?

File: src/janetsh [hunk 1]
```diff
@@ -60,9 +60,21 @@
...
 (array/concat process/args new-args)
 
+(defn cwd-tilded
+  "Returns current directory with ~ in place of $HOME."
+  []
+  (let [cwd (os/cwd)
+        home (os/getenv "HOME")
+        dir (cond
+              (string/has-prefix? (string home "/") cwd)
+              (string/replace home "~" cwd)
```
Comment (andrewchambers) [inline src/janetsh:70]: String replace is a bug - you need to only replace the first tild.

 I think this code can be shortened a tad using an or and string/slice, you only need two conditions.

**Review by andrewchambers — CHANGES_REQUESTED** [review]

**Pull request merged**